File: actionhero-backend/src/models/Company.ts
import mongoose, { Document, Schema } from 'mongoose';
import { IJobOpportunity } from './JobOpportunities'

export interface ICompany extends Document {
  name: string;
  numberOfEmployees: number;
  email: string;
  phoneNumber: string;
  description: string;
  jobOpportunities?: IJobOpportunity[];
}

const CompanySchema: Schema<ICompany> = new Schema({
  name: { type: String, required: true },
  numberOfEmployees: { type: Number, required: true },
  email: { type: String, required: true },
  phoneNumber: { type: String, required: true },
  description: { type: String, required: true },
});

export const Company = mongoose.model<ICompany>('Company', CompanySchema);

File: resumate_flutter/lib/features/feed/model/Company.dart
class Company {
  final String id;
  final String name;
  final int numberOfEmployees;
  final String email;
  final String phoneNumber;
  final String description;
  final int v;

  Company({
    required this.id,
    required this.name,
    required this.numberOfEmployees,
    required this.email,
    required this.phoneNumber,
    required this.description,
    required this.v,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['_id'],
      name: json['name'],
      numberOfEmployees: json['numberOfEmployees'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      description: json['description'],
      v: json['__v'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'numberOfEmployees': numberOfEmployees,
      'email': email,
      'phoneNumber': phoneNumber,
      'description': description,
      '__v': v,
    };
  }
}
